ui-components/src/helpers/pdfs.ts

```diff
@@ -0,0 +1,14 @@
+import { ListingEvent, ListingEventType } from "@bloom-housing/backend-core/types"
+
+export const cloudinaryPdfFromId = (publicId: string) => {
+  const cloudName = process.env.cloudinaryCloudName || process.env.CLOUDINARY_CLOUD_NAME
+  return `https://res.cloudinary.com/${cloudName}/image/upload/${publicId}.pdf`
+}
+
+export const pdfUrlFromListingEvents = (events: ListingEvent[], listingType: ListingEventType) => {
+  const event = events.find((event) => event.type === listingType)
+  if (event) {
+    return event.file?.label == "cloudinaryPDF" ? cloudinaryPdfFromId(event.file.fileId) : event.url
+  }
+  return null
+}
```

Comment:
Is there a decent way we can make this generic (not dependent on backend/core/types)? If not, due to the more recent efforts started with SFDS, I'm not sure we want it in UI Components.

Reply:
It's definitely easiest to keep this logic contained within the single function. We could duplicate it across both partners and public but I'm not sure if that's a good idea either. Is it time to bite the bullet and create a "shared helpers" sort of package?

Reply:
I think we should, but we don't need to do that here. I'll create a separate issue for it.